toolslm/download.py

```diff
@@ -90,8 +91,11 @@ def find_docs(url):
 # %% ../03_download.ipynb 19
 def read_docs(url, optional=False, n_workers=None, rm_comments=True, rm_details=True):
     "If available, return LLM-friendly llms.txt context or markdown file response for `url`"
-    url = find_docs(url)
-    if not url: return
-    if url.endswith('/llms.txt'): res = get_llmstxt(url, optional=optional, n_workers=n_workers)
-    else: res = get(url).text
+    fn = find_docs(url)
+    if not fn:
+        parsed_url = urlparse(url)
+        if parsed_url.path == '/' or not parsed_url.path: return None
+        return read_docs(urljoin(url, '..'), optional, n_workers, rm_comments, rm_details)
+    if fn.endswith('/llms.txt'): res = get_llmstxt(fn, optional=optional, n_workers=n_workers)
```

Comment:
By having this after the parent search, does that mean we're not finding llms.txt in parents?

Reply:
> By having this after the parent search, does that mean we're not finding llms.txt in parents?

the parent search involves a recursive function call, in which case `read_docs` will be called on `url:parent`. If there's an `fn` found it will hit line 99.